### ISSUE TITLE
[BrowserGestureDetector] Crash in [@ java.lang.IllegalArgumentException: at kotlin.jvm.internal.ArrayIteratorKt.throwParameterIsNullException(ArrayIterator.kt:5)]

### ISSUE THREAD

**Issue by kbrosnan** [mozilla-mobile/android-components#8356]
Crash report: https://crash-stats.mozilla.org/report/index/3e85c676-c77c-4ca8-ab56-1504d0200908

Java stack trace:
```
java.lang.IllegalArgumentException
	at kotlin.jvm.internal.ArrayIteratorKt.throwParameterIsNullException(ArrayIterator.kt:5)
	at kotlin.jvm.internal.ArrayIteratorKt.checkParameterIsNotNull(ArrayIterator.kt:1)
	at mozilla.components.browser.toolbar.behavior.BrowserGestureDetector$CustomScrollDetectorListener.onScroll(Unknown Source:2)
	at android.view.GestureDetector.onTouchEvent(GestureDetector.java:625)
	at mozilla.components.browser.toolbar.behavior.BrowserGestureDetector.handleTouchEvent$browser_toolbar_release(BrowserGestureDetector.kt:7)
	at mozilla.components.browser.toolbar.behavior.BrowserToolbarBottomBehavior.onInterceptTouchEvent(BrowserToolbarBottomBehavior.kt:2)
	at mozilla.components.browser.toolbar.behavior.BrowserToolbarBottomBehavior.onInterceptTouchEvent(BrowserToolbarBottomBehavior.kt:1)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.performIntercept(CoordinatorLayout.java:13)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onInterceptTouchEvent(CoordinatorLayout.java:3)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2573)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3030)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2719)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:452)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1836)
	at android.app.Activity.dispatchTouchEvent(Activity.java:3400)
	at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:1)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:410)
	at android.view.View.dispatchPointerEvent(View.java:12768)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5127)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4930)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4447)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4500)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4466)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4606)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4474)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4663)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4447)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4500)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4466)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4474)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4447)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7124)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7093)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7054)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7227)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:187)
	at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
	at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:178)
	at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:7198)
	at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:7250)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1004)
	at android.view.Choreographer.doCallbacks(Choreographer.java:816)
	at android.view.Choreographer.doFrame(Choreographer.java:745)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:990)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:280)
	at android.app.ActivityThread.main(ActivityThread.java:6706)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by pocmo** [comment]
CC @Mugurell

**Comment by Mugurell** [comment]
Thank you Sebastian!

Based on
> java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.ArrayIteratorKt.checkParameterIsNotNull, parameter e1
	at kotlin.jvm.internal.ArrayIteratorKt.throwParameterIsNullException(ArrayIterator.kt:5)
	at kotlin.jvm.internal.ArrayIteratorKt.checkParameterIsNotNull(ArrayIterator.kt:1)
	at mozilla.components.browser.toolbar.behavior.BrowserGestureDetector$CustomScrollDetectorListener.onScroll(Unknown Source:2)
	at android.view.GestureDetector.onTouchEvent(GestureDetector.java:639)

I understand that Android's `GestureDetector` calls us as a listener with null parameters although it's method definition says the elements are non-nullable. And that first parameter should've been the `ACTION_DOWN` event.
In testing this I could not reproduce but I think we're seeing cases in which some MotionEvents are not forwarded to the GestureDetector. Otherwise it might be a bug in the GestureDetector.

As a speculative fix I'll 
- refactor out the `by lazy(LazyThreadSafetyMode.NONE)` GestureDetector initialization. Though all calls should've been on a single Thread and so ensure serial execution.. just to make sure we don't end up with more detectors and somehow some MotionEvents go to one detector and others to another.
- ensure we always pass `ACTION_DOWN`, `ACTION_UP`, `ACTION_CANCEL` to the detector.

**Comment by jonalmeida** [comment]
Closing since this the PR for this was merged in 62.0.0
